climpred/relative_entropy.py

```diff
@@ -0,0 +1,291 @@
+    for _ in range(it):
+        control_uninitialized = _pseudo_ens(ds, control)
+        control_uninitialized['initialization'] = ds.initialization.values
+        ds_list.append(control_uninitialized)
+    control_uninitialized = xr.concat(ds_list, 'member')
+    control_uninitialized['member'] = np.arange(control_uninitialized.member.size)
+    return control_uninitialized
+
+
+def _gen_control_distribution_lens(ds, control, max_member=40):
+    """
+    Generate a large control distribution from control, LENS like.
+
+    Args:
+        ds (xr.DataArray): initialization data with dimensions initialization,
+                           member, time and spatial [lon (x), lat(y)].
+        control (xr.DataArray): control data with dimensions time and spatial
+                                [lon (x), lat(y)].
+        it (int): multiplying factor for ds.member.
+
+    Returns:
+        control_uninitialized (xr.DataArray): data with dimensions member, time
+                                              and spatial [lon (x), lat(y)].
+
+    """
+    ds_list = []
+    for _ in range(int(max_member/ds.member.size)+1):
+        control_uninitialized = _pseudo_ens(ds, control)
+        control_uninitialized['initialization'] = ds.initialization.values
+        ds_list.append(control_uninitialized)
+    control_uninitialized = xr.concat(ds_list, 'member')
+    control_uninitialized['member'] = np.arange(control_uninitialized.member.size)
+    return control_uninitialized.isel(
+                    initialization=0,member=slice(0,40)).drop('initialization')
+
+
+def compute_relative_entropy(initialized, control_uninitialized,
+                             anomaly_data=False, neofs=None, curv=True,
+                             ntime=None, detrend_by_control_unitialized=True):
+    """
+    Compute relative entropy.
+
+    Calculates EOFs from anomalies. Projects fields on EOFs to receive
+    pseudo-Principle Components per initialization and lead year. Calculate
+    relative entropy based on _relative_entropy_formula.
+
+    Args:
+        initialized (xr.DataArray): anomaly ensemble data with dimensions
+                                    initialization, member, time and spatial
+                                    [lon (x), lat(y)].
+        control_uninitialized (xr.DataArray): anomaly control distribution with
+                                              dimensions init, member, time and
+                                              spatial [lon (x), lat(y)].
+        anomaly_data (bool): Input data is anomaly alread. Default: False.
+        neofs (int): number of EOFs to use. Default: initialized.member.size.
+        curv (bool): if curvilinear grids are provided disables EOF weights.
+        ntime (int): number of timesteps calculated.
+        detrend_by_control_unitialized (bool): Default: True
+
+    Returns:
+        rel_ent (pd.DataFrame): relative entropy
+
+    """
+    if neofs is None:
+        neofs = initialized.member.size
```

Comment:
Shouldn't this default to `initialized.member.size - 1` based on the theory and notebook you had?

Reply:
implementation-wise ds.member.size is the limit, however I dont really understand which number of EOFs should be taken... 5, 15 all quite random. we could check how much variance these EOFs produce but still its a choice.